numpy/lib/stride_tricks.py

```diff
@@ -111,6 +111,145 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
 
     return view
 
+def sliding_window_view(x, shape, step=None, subok=False, writeable=False):
+    """
+    Create sliding window views of the N dimensions array with the given window
+    shape. Window slides across each dimension of `x` and extract subsets of `x`
+    at any window position.
+
+    Parameters
+    ----------
+    x : ndarray
+        Array to create sliding window views.
+
+    shape : sequence of int
+        The shape of the window. Must have same length as number of input array dimensions.
+
+    step: sequence of int, optional
+        The steps of window shifts for each dimension on input array at a time.
+        If given, must have same length as number of input array dimensions.
+        Defaults to 1 on all dimensions.
+
+    subok : bool, optional
+        If True, then sub-classes will be passed-through, otherwise the returned
+        array will be forced to be a base-class array (default).
+
+    writeable : bool, optional
+        If set to False, the returned array will always be readonly view.
+        Otherwise it will return writable copies(see Notes).
+
+    Returns
+    -------
+    view : ndarray
+        Sliding window views (or copies) of `x`. view.shape = (x.shape - shape) // step + 1
+
+    See also
+    --------
+    as_strided: Create a view into the array with the given shape and strides.
+    broadcast_to: broadcast an array to a given shape.
+
+    Notes
+    -----
+    ``sliding_window_view`` create sliding window views of the N dimensions array
+    with the given window shape and its implementation based on ``as_strided``.
+    Please note that if writeable set to False, the return is views, not copies
+    of array. In this case, write operations could be unpredictable, so the return
+    views is readonly. Bear in mind, return copies (writeable=True), could possibly
+    take memory multiple amount of origin array, due to overlapping windows.
+
+    For some cases, there may be more efficient approaches, such as FFT based algo discussed in #7753.
```

Comment:
I don't like referring to github discussions from within NumPy documentation. Perhaps this is better?

For some cases there may be more efficient approaches to calculate transformations across multi-dimensional arrays, for instance `scipy.signal.fftconvolve`, where combining the iterating step with the calculation itself while storing partial results can result in significant speedups.